common/src/main/java/org/dash/wallet/common/WalletDataProvider.kt

```diff
@@ -30,6 +30,7 @@ import org.dash.wallet.common.transactions.TransactionFilter
 import org.dash.wallet.common.transactions.TransactionWrapper
 
 interface WalletDataProvider {
+    val wallet: Wallet?
```

Comment:
The original idea behind `WalletDataProvider` was to give other modules some data without providing the Wallet itself.  Probably some of the reason for this was that third parties may be writing these integrations (which was done in the case of Liquid).  The `Wallet` object would be fully owned by the `wallet` module (which includes the means to decrypt the keys).

Reply:
Yes, this is a correct idea and we should keep to it (don't use `wallet` object in the feature modules). Why I have put it in there for now:
- We have a lot of places that use the `WalletApplication` while they shouldn't. This is a god-object that does too much and provides a lot of access to a lot of functions. It's too tempting to call `getInstance()` on it and break all of the DI patterns we have added. Just now I had to deal with this problem while writing tests for the Explore database, even if injections were used there for some other things.
- Most places only need the `wallet`, not the entire `WalletApplication`.
- We need to abstract the access to the `wallet` from the `WalletApplication` somehow. I think the best option here would be to have some kind of "wallet storage" class that is getting updated by the `WalletApplication`. This class would only be available from the `wallet` module.
- Until this solution exists (I'm not 100% sure yet on what it should look like), we need to keep abstracting the access to `wallet` from the `WalletApplication`. I suppose we could create yet another interface for it, but I fear this could be even more confusing.

So for now I figure we can have the `wallet` in this interface, and once the access to it is moved from the `WalletApplication`, we will simply remove it from here and replace it with another injection which should be relatively easy to do.

Reply:
I've added a comment warning from using it in the feature modules.